backend-zs/routes/story.routes.js

```diff
@@ -6,6 +6,9 @@ function storyRoutes() {
 
     router.get('/', storiesController.getStories)
     router.get('/:id', storiesController.getStory)
+    router.delete('/:id', storiesController.deleteStory)
+    router.post('/', storiesController.createStory)
+    router.post('/update', storiesController.updateStory)
```

Comment:
Middlewares can be used for checking if:

- creating user is an admin, authorImage belongs to the user and exists in S3
- deleting user is an admin and is the owner of the story, authorImage(to be deleted) belongs to the user
- updating user is an admin and is the owner of the story, and authorImage (if present in the request) belongs to the user

All middlewares already exist in `projects.controller` and `users.controller`.

Reply:
I have added `verifyUserIsAdmin` to create and delete, because: 

- creating: 'authorImage belongs to user and exists in S3' will not hold because we are uploading during the creation process
- deleting: a story does not have an owner, also I think that each admin should be able to delete the story not just the ones who uploaded the authorImage
- updating: the update API is used when the user clicked the button like, all users can click the like

Reply:
>updating: the update API is used when the user clicked the button like, all users can click the like

For this, I'd prefer to have an independent endpoint for the like. Otherwise anyone would be able to edit the story data.

If you prefer fixing this in a next iteration during the feature freeze, this would be ok (but In my opinion it's a must to be there).

Reply:
You are right. I implemented it this way. I also kept the normal update api for admins.